rust/src/lib.rs

```diff
@@ -17,6 +17,66 @@
 //! Lance columnar data format is an alternative to Parquet. It provides 100x faster for random access,
 //! automatic versioning, optimized for computer vision, bioinformatics, spatial and ML data.
 //! [Apache Arrow](https://arrow.apache.org/) and DuckDB compatible.
+//!
+//!
+//! # Create a Dataset
+//!
+//! ```rust
+//! # use std::sync::Arc;
+//! # use tokio::runtime::Runtime;
+//! # use arrow_array::{RecordBatch, RecordBatchIterator};
+//! # use arrow_schema::{Schema, Field, DataType};
+//! use lance::{dataset::WriteParams, Dataset};
+//!
+//! # let mut rt = Runtime::new().unwrap();
+//! # rt.block_on(async {
+//! #
+//! # let schema = Arc::new(Schema::new(vec![Field::new("test", DataType::Int64, false)]));
+//! # let batches = vec![RecordBatch::new_empty(schema.clone())];
+//! # let test_dir = tempfile::tempdir().unwrap();
+//! # let uri = test_dir.path().to_str().unwrap().to_string();
+//! let write_params = WriteParams::default();
+//! let reader = RecordBatchIterator::new(
+//!     batches.into_iter().map(Ok), schema
+//! );
+//! Dataset::write(reader, &uri, Some(write_params)).await.unwrap();
```

Comment:
I feel like we should show a little more detail on how to make a RBR:

```suggestion
//! # let test_dir = tempfile::tempdir().unwrap();
//! # let uri = test_dir.path().to_str().unwrap().to_string();
//! let schema = Arc::new(Schema::new(vec![Field::new("test", DataType::Int64, false)]));
//! let batches = vec![RecordBatch::new_empty(schema.clone())];
//! let reader = RecordBatchIterator::new(
//!     batches.into_iter().map(Ok), schema
//! );
//!
//! let write_params = WriteParams::default();
//! Dataset::write(reader, &uri, Some(write_params)).await.unwrap();
```